samza-log4j/src/main/java/org/apache/samza/logging/log4j/StreamAppender.java

```diff
@@ -206,13 +259,51 @@ protected void setupSystem() {
 
     setSerde(log4jSystemConfig, systemName, streamName);
 
-    systemProducer = systemFactory.getProducer(systemName, config, new MetricsRegistryMap());
+    systemProducer = systemFactory.getProducer(systemName, config, metricsRegistry);
```

Comment:
A quick question: here we are creating a new producer for StreamAppender, which is not part of the SystemProducers. Hence, not controlled by collector.flush(). For logs, I assume that we would like systemProducer to be highly available, more than no-loss or order-guarantee, as we configure for input/output systems. Can we auto-configure this systemProducer to drop producer errors?

Reply:
Great idea:
https://issues.apache.org/jira/browse/SAMZA-1540